MohammadBallal/task.py

```diff
@@ -0,0 +1,40 @@
+class Task:
+    idGenerator= 0
+    def __init__(self,title,description,due_date) :
+        #Initialize the attributes
+        Task.idGenerator+= 1
+        self.task_id= Task.idGenerator
+        self.title= title
+        self.description= description
+        self.due_date= due_date
+        self.status="InProgress"
```

Comment:
That's the thing – someone might create a list of tasks without starting to work on it or  It might happen that they want to create a group of completed tasks so they can remember that they have already completed the task (you can think of it as a note).

Reply:
I understand you, as you mentioned this scenario might happen and for this there is an update_task function, but in most cases tasks usually start as 'InProgress' and for this reason using default value.

Reply:
Why should we do it in two steps instead of one? I mean, create, then update. Another thing to consider is that when we add a new status, let's say status 'create', which means we need to update two places instead of simply adding that status to the supported list. Why should we impose this limitation? It should be up to the user to add the status we want.

Reply:
Yes, you're right. 
And about adding a new state, I was restricted by the Python task to have only two states. As I mentioned, in most cases tasks usually start in the 'In Progress' state.
I'll modify the code now.

Reply:
Done